src/api/Controllers/BoreholeGeometryController.cs

```diff
@@ -0,0 +1,317 @@
+    public async Task<IActionResult> DeleteAsync(int boreholeId)
+    {
+        // Check if associated borehole is locked
+        if (await boreholeLockService.IsBoreholeLockedAsync(boreholeId, HttpContext.GetUserSubjectId()).ConfigureAwait(false))
+        {
+            return Problem("The borehole is locked by another user or you are missing permissions.", statusCode: (int)HttpStatusCode.BadRequest);
+        }
+
+        context.BoreholeGeometry.RemoveRange(context.BoreholeGeometry.Where(g => g.BoreholeId == boreholeId));
+        await context.SaveChangesAsync().ConfigureAwait(false);
+        return Ok();
+    }
+
+    [HttpGet("[action]")]
+    [Authorize(Policy = PolicyNames.Viewer)]
+    public IList<object> GeometryFormats()
+    {
+        return geometryFormats.Select(f => (object)new
+        {
+            f.Name,
+            f.Key,
+            f.CsvHeader,
+        }).ToList();
+    }
+
+    [HttpPost]
+    [Authorize(Policy = PolicyNames.Viewer)]
+    [RequestSizeLimit(int.MaxValue)]
+    [RequestFormLimits(MultipartBodyLengthLimit = MaxFileSize)]
+    public async Task<IActionResult> UploadBoreholeGeometry(int boreholeId, IFormFile geometryFile, [FromForm] string geometryFormat)
+    {
+        // Check if associated borehole is locked
+        if (await boreholeLockService.IsBoreholeLockedAsync(boreholeId, HttpContext.GetUserSubjectId()).ConfigureAwait(false))
+        {
+            return Problem("The borehole is locked by another user or you are missing permissions.", statusCode: (int)HttpStatusCode.BadRequest);
+        }
+
+        var format = geometryFormats.SingleOrDefault(f => f.Key == geometryFormat);
+        if (format == null)
+        {
+            return Problem("Invalid geometry format.", statusCode: (int)HttpStatusCode.BadRequest);
+        }
+
+        // convert geometry data to BoreholeGeometry
+        IList<BoreholeGeometry> boreholeGeometry;
+        try
+        {
+            boreholeGeometry = format.ReadCsv(geometryFile, boreholeId);
+        }
+        catch (Exception ex) when (ex is HeaderValidationException || ex is ReaderException || ex is TypeConverterException)
+        {
+            return Problem(ex.Message, statusCode: (int)HttpStatusCode.BadRequest);
+        }
+
+        // delete existing geometry data of borehole
+        context.BoreholeGeometry.RemoveRange(context.BoreholeGeometry.Where(g => g.BoreholeId == boreholeId));
+
+        // add new geometry data to database
+        await context.BoreholeGeometry.AddRangeAsync(boreholeGeometry).ConfigureAwait(false);
+
+        try
+        {
+            await context.SaveChangesAsync().ConfigureAwait(false);
+            return Ok();
+        }
+        catch (Exception ex)
+        {
+            var errorMessage = "An error occurred while saving the borehole geometry.";
+            logger?.LogError(ex, errorMessage);
+            return Problem(errorMessage);
+        }
+    }
+
+    private interface IBoreholeGeometryFormat
```

Comment:
Q: Warum ist das interface private? Sonst haben wir die Interfaces ja meistens in eigenen Files, ist das hier nicht sinnvoll ?

Reply:
Wenn das Interface nicht private ist muss es in ein eigenes File. Ich hatte das gefühl, das die BoreholeGeometryController  Logik nur unnötig verstreut wird, wenn sie in verschiedene Files ist. Das Interface habe ich gemacht, damit die Umrechnungen einfacher erweiterbar und verständlich sind.